Sources/Misc/MapAppStoreDetector.swift

```diff
@@ -32,6 +32,8 @@ final class DefaultMacAppStoreDetector: MacAppStoreDetector {
     /// it checked the common name but was changed to an extension check to make it more
     /// future-proof.
     ///
+    /// - warning: this method may take a long time to run on macOS. It should not be called on the main thread. 
```

Comment:
these warnings should ideally be fixed by the solutions described in the PR description, but I couldn't find a great way to make them work, sadly